test/src/sender/main.cpp

```diff
@@ -1,141 +1,186 @@
+struct inp_controll : public cadmium::in_port<Message_t> {};
+struct inp_ack : public cadmium::in_port<Message_t> {};
+
+/***** SETING OUTPUT PORTS FOR COUPLEDs *****/
+struct outp_ack : public cadmium::out_port<Message_t> {};
+struct outp_data : public cadmium::out_port<Message_t> {};
+struct outp_pack : public cadmium::out_port<Message_t> {};
+
+
+/********************************************/
+/****** APPLICATION GENERATOR *******************/
+/********************************************/
+template<typename T>
+class ApplicationGen : public iestream_input<Message_t,T> {
+    public:
+        ApplicationGen() = default;
+        ApplicationGen(const char* file_path) :
+            iestream_input<Message_t,T>(file_path) {
+    }
+};
+
+
+int main() {
+
+    auto start = hclock::now(); //to measure simulation execution time
+
+    /*************** Loggers *******************/
+    static std::ofstream out_data(
+        "../test/data/sender/sender_test_output.txt");
+    struct oss_sink_provider {
+        static std::ostream& sink() {
+            return out_data;
+        }
+    };
+
+    using
+        info = cadmium::logger::logger<cadmium::logger::logger_info,
+            cadmium::dynamic::logger::formatter<TIME>,
+                oss_sink_provider>;
+    using
+        debug = cadmium::logger::logger<cadmium::logger::logger_debug,
+            cadmium::dynamic::logger::formatter<TIME>,
+                oss_sink_provider>;
+    using
+        state = cadmium::logger::logger<cadmium::logger::logger_state,
+            cadmium::dynamic::logger::formatter<TIME>,
+                oss_sink_provider>;
+    using
+        log_messages = cadmium::logger::logger<cadmium::logger::logger_messages,
+            cadmium::dynamic::logger::formatter<TIME>,
+                oss_sink_provider>;
+    using
+        routing = cadmium::logger::logger<cadmium::logger::logger_message_routing,
+            cadmium::dynamic::logger::formatter<TIME>,
+                oss_sink_provider>;
+    using
+        global_time = cadmium::logger::logger<cadmium::logger::logger_global_time,
+            cadmium::dynamic::logger::formatter<TIME>,
+                oss_sink_provider>;
+    using
+        local_time = cadmium::logger::logger<cadmium::logger::logger_local_time,
+            cadmium::dynamic::logger::formatter<TIME>,
+                oss_sink_provider>;
+    using
+        log_all = cadmium::logger::multilogger<info,
+            debug, state, log_messages, routing, global_time, local_time>;
+
+    using logger_top = cadmium::logger::multilogger<log_messages, global_time>;
+
+
+    /*******************************************/
+
+
+
+    /********************************************/
+    /****** APPLICATION GENERATOR *******************/
+    /********************************************/
+    string input_data_control =
+        "../test/data/sender/sender_input_test_control_In.txt";
+    const char * i_input_data_control = input_data_control.c_str();

```

Comment:
there wont be any space between char and *

Reply:
solved